File: examples/basic.rs
// 基本命令执行示例：创建沙箱并执行一条简单命令。
#[cfg(all(feature = "os", any(target_os = "linux", target_os = "macos")))]
use mimobox_sdk::Sandbox;

#[cfg(all(feature = "os", any(target_os = "linux", target_os = "macos")))]
fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mut sandbox = Sandbox::new()?;
    let result = sandbox.execute("/bin/echo hello mimobox")?;
    println!("exit: {:?}", result.exit_code);
    println!("stdout: {}", String::from_utf8_lossy(&result.stdout));
    sandbox.destroy()?;
    Ok(())
}

#[cfg(not(all(feature = "os", any(target_os = "linux", target_os = "macos"))))]
fn main() {
    eprintln!("此示例需要 Linux/macOS + mimobox-sdk 的 os feature。");
}

File: examples/python/basic.py
# basic.py — Minimal example: create sandbox, execute command, print result, close
#
# Demonstrates the most basic Sandbox usage: without context manager, manual lifecycle management.

from mimobox import Sandbox

# Create sandbox (using default auto isolation level)
sandbox = Sandbox()

# Execute command
result = sandbox.execute("echo 'Hello from mimobox!'")

# Print result
print(f"exit_code: {result.exit_code}")
print(f"stdout:    {result.stdout.strip()}")
print(f"stderr:    {result.stderr.strip()}")
print(f"timed_out: {result.timed_out}")
if result.elapsed is not None:
    print(f"elapsed:   {result.elapsed:.3f}s")

# Manual cleanup: close() is the recommended cleanup method when not using a with statement.
sandbox.close()
print("sandbox closed")
